### PR TITLE
fixes #24130 - use new rendering engine

### PR DESCRIPTION
**Pull request by timogoebel**
This needs theforeman/foreman#5683.

### CODE REVIEW

File: app/lib/foreman_remote_execution/renderer/input_template_scope.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+module ForemanRemoteExecution
```
Comment (timogoebel) [inline app/lib/foreman_remote_execution/renderer/input_template_scope.rb]: We don't need this actually.

**Comment by iNecas** [comment]
Thanks for looking into this. During the first round of testing, I've noticed in editing some templates (such as Package Action - SSH Default), the preview fails with:

```
Warning! Template is missing Missing template job_templates/preview, templates/preview, application/preview with {:locale=>[:en], :formats=>[:html, :text, :js, :css, :ics, :csv, :vcf, :png, :jpeg, :gif, :bmp, :tiff, :svg, :mpeg, :xml, :rss, :atom, :yaml, :multipart_form, :url_encoded_form, :json, :pdf, :zip, :gzip], :variants=>[], :handlers=>[:raw, :erb, :html, :builder, :ruby, :rabl]}. Searched in: * "/home/inecas/Projects/ws/foreman-rex/foreman-rails5/app/views" * "/home/inecas/Projects/ws/foreman-rex/foreman_ansible/app/views" * "/home/inecas/Projects/ws/foreman-rex/foreman_remote_execution/app/views" * "/home/inecas/Projects/ws/foreman-rex/foreman-tasks/app/views" * "/home/inecas/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/apipie-rails-0.5.9/app/views"
```

I didn't have a chance to debug this more deeply yet.

**Comment by timogoebel** [comment]
> I didn't have a chance to debug this more deeply yet.

It looks like this has been broken for a while (I suggest the rails upgrade).
There is not a single test for the controller, so it is no surprise this wasn't noticed earlier.

The actual error is that `operatingsystem` is undefined for the scope. I still need to investigate this.
But first, I'll add a couple of controller tests.

**Comment by timogoebel** [comment]
@iNecas: I forgot to pass `@host` as a variable to the renderer, that's why `@host.operatingsystem` showed an error.
The error message you saw is because `render text:` does not work anymore with Rails 5, it has to be `render plain:`. I fixed that as well and added some basic tests for `#preview` in templates controller.

File: app/models/input_template_renderer.rb [hunk 2]
```diff
@@ -16,84 +16,64 @@ def initialize(template, host = nil, invocation = nil, input_values = nil, previ
...
+    @invocation = invocation
     @input_values = input_values
     @preview = preview
     @templates_stack = templates_stack + [template]
   end
 
   def render
     @template.validate_unique_inputs!
-    render_safe(@template.template, ::Foreman::Renderer::ALLOWED_HELPERS + [ :input, :render_template, :preview?, :render_error ], :host => @host)
+    Foreman::Renderer.render_template(
```
Comment (iNecas) [inline app/models/input_template_renderer.rb]: With the latest version of https://github.com/theforeman/foreman/pull/5683/files, I'm getting:

```
Failed to initialize: RuntimeError - Failed rendering template: error during rendering: undefined method `render_template' for Foreman::Renderer:Module
   2:
Did you mean? render_template_to_tempfile
```

**Comment by timogoebel** [comment]
Rebased and updated to work with the latest PR code.

**Comment by iNecas** [comment]
Tested and works well: ready to merge after the core one is in.

**Comment by iNecas** [comment]
Running tests locally, they also pass successfully.

**Comment by iNecas** [comment]
[test foreman_remote_execution]

**Comment by iNecas** [comment]
Thanks @timogoebel for great work.

**Comment by iNecas** [comment]
foreman-rex 1.6.0 has been pushed to rubygems

**Pull request merged**